src/main/java/io/english/controller/UsersController.java

```diff
@@ -45,8 +45,29 @@
 
     @PostMapping
     public ResponseEntity<UserResponse> create(@Valid @RequestBody UserCreateRequest request) {
-        User user = userService.create(request);
+        var user = userService.create(request);
         UserResponse response = UserMapper.INSTANCE.toResponse(user);
         return ResponseEntity.ok(response);
     }
+
+    @PutMapping("{id}/teacher")
```

Comment:
PUT /users/{id}/teacher would rather mean "user with id {id} do something with his teacher", which contradicts the logic of the endpoint
Maybe it should be /users/current/teacher/{id} which means "current user do something with his teacher that has id {id}" or come up with better url